src/check-reserve-update-frequency/check-reserve-update-frequency.js

```
@@ -0,0 +1,166 @@
+  );
+
+  // get an array of all of the reserve tokens, in the form of TokenData struct entries
+  // ref: https://docs.aave.com/developers/the-core-protocol/protocol-data-provider#getallreservestokens
+  let reserveTokenArray = await protocolDataProviderContract.getAllReservesTokens();
+
+  // remove tokens that do not have Chainlink price oracles
+  reserveTokenArray = reserveTokenArray.filter(
+    (reserveToken) => TOKENS_WITHOUT_CHAINLINK_ABI.indexOf(reserveToken.tokenAddress) === -1,
+  );
+
+  // get the AAVE price oracle address
+  const priceOracleAddress = await lendingPoolAddressProviderContract.getPriceOracle();
+  const priceOracleContractInstance = new ethers.Contract(
+    priceOracleAddress,
+    priceOracleAbi,
+    jsonRpcProvider,
+  );
+
+  // get the price source addresses
+  const priceSourceAddresses = await Promise.all(reserveTokenArray.map(
+    (reserveToken) => priceOracleContractInstance.getSourceOfAsset(reserveToken.tokenAddress),
+  ));
+
+  // create ethers contracts to run read-only methods from the Chainlink contracts
+  const priceSourceContractInstances = priceSourceAddresses.map(
+    (priceSourceAddress) => new ethers.Contract(
+      priceSourceAddress,
+      chainlinkAggregatorAbi,
+      jsonRpcProvider,
+    ),
+  );
+
+  // create an array of token / address / contract tuples that we will iterate over
+  const tokenAddressContractTuples = reserveTokenArray.map((token, index) => {
+    const address = priceSourceAddresses[index];
+    const contract = priceSourceContractInstances[index];
+    return [token, address, contract];
+  });
+
+  return tokenAddressContractTuples;
+}
+
+function provideHandleBlock(tokensAddressesContractsPromise) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    // settle the promise the first time, all subsequent times just get the resolve() value
+    const tokensAddressesContracts = await tokensAddressesContractsPromise;
```

no error checking on this, do we care? I think no :) up to you

---

I agree that if this fails, we should just let it fail.